### PR TITLE
[NEED NUMBER TWEAKING?] Surgeries now give techweb points on completion.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: XDTM (hijacked by Cobby)
add: Surgeries now give points on completion. A list SHOULD be on the wiki
/:cl:

closes #40158
Purpose: to give medbay a way to contribute to the cooperative progression system.

**I do not know how I'd display this information ingame, and I need review on the numbers. I've put disks ones high and farmable ones super low then tried to make the others relative to that.**

Surgery | Point Amount
-- | --
Bioware Surgeries | 3000
Brainwashing | 3000
Lobotomy | 750 per trauma
Necrotic | 7500
Pacification | 3000
Reconstruction | 2500
Revival | 5000
Viral Bond | 3000
Amput | 500
Brain Surgery | 500 per trauma
Cavity | 100
Core Removal | 100 per extract
Dental | 100
Eye Surgery | 25*eye damage, capped at 725
Implant Removal | 100
Augment | 1000
Lipo | 500
Plastic Surgery | 500
Prosthetics | 750
Remove Embed | 500

### CODE REVIEW

**Comment by Poojawa** [comment]
Ensure corpses must have had a ckey in them or you're going to get mutagen farmers cheesing the system, tbh.

**Comment by Poojawa** [comment]
Passive generation was only so high because there was basically nothing to generate points.

with more features added, like anomaly cores, engine set ups, etc, allows for the passive gen to get reduced. It shouldn't be the benchmark to judge a worthiness.

**Comment by XDTM** [comment]
If you want to address what @kevinz000 said, if you restrict point gain to only advanced surgeries you'll also force being near a console.

**Comment by XDTM** [comment]
Also most of them are repeatable, so you probably want much lower amount for those.

**Comment by ExcessiveUseOfCobblestone** [comment]
Doing ghetto surgery in a backroom for "research" sounds so ss13 though.

*pulls out glass shard* don't worry this is for science.

@Tlaltecuhtli I didn't base it off passive gen, I based it off of typical tech costs (2500)

with a very loose definition of "base".

@Poojawa it requires a client check at least by the end, so you can dc until the final step :^)

**Comment by RogueSteampunker** [comment]
I mean, it checks out in my book. Robotics and Med can both do this, all the major hitters are genuine one-time things that you don't replicate easily, and the smaller ones are literally enough work that, if you're going to specifically load a guy up with chems just so you can pneumatic cannon a bunch of shards into his skin for those dank embedded points, I think you deserve em.

**Comment by Poojawa** [comment]
If the corpse has had a Ckey at all, it's valid. s'what I'm sayin'. 

My point is for genetics monkey men or NPC corpses who have never had a Ckey shouldn't contribute, or at the very most contribute much, much less.

**Comment by ExcessiveUseOfCobblestone** [comment]
Outside of disconnects you shouldn't have someone start /w ckey then be ckeyless by the time the surgery is over.

If someone disconnects just do all but the last part.

**Comment by Poojawa** [comment]
Except, y'know. People in genetics who toggle the Monkey -> Human gene and make a "monkey man" or 'unmonkey'.

Or NPCs that die and spawn a random human with all the equipment on it.

**Comment by davethwave** [comment]
One easy way to take advantage of this is Impedrezene and lobotomy/brain surgery, inject 15u every time the person is cured of trauma.

**Comment by XDTM** [comment]
or just spam reconstruction

**Comment by ExcessiveUseOfCobblestone** [comment]
@Poojawa what does flipping species have to do with ckeys? when does NPCs start with ckeys?

What?

I already have a client check, they won't be beneficial to the process.

@davethwave @XDTM would you knowingly drink something that's going to kick your but then undergo surgery the entire round just for techweb points?

**Comment by Poojawa** [comment]
Client check != ckey check

You won't be able to perform this on dead, cloned people. So someone can just see that they're being pacified, amputated, and whatever else and just log out/ghost.

but if the corpse had a ckey in it at all, then you can still perform what amounts to post-mortium autopsies to make use of cloned people's bodies without allowing NPC bodies to be used.

That's what I'm trying to get at, sorry if I wasn't being clear.

**Comment by davethwave** [comment]
Their are usually a few braindead people ingame as well you can experiment on.

**Comment by Poojawa** [comment]
Most servers tend to have a 'don't fuck with braindead people unless you have to' rule.

**Comment by ExcessiveUseOfCobblestone** [comment]
Therapist1: It's because this is doing your job (or doing epic back-alley stuff) vs going out of your way to specifically do a certain surgery.

I tried to make it synonymous with the other active gen method: toxins. Obviously surgery is not calculation based so the next best thing is amount of surgeries done.

**Comment by XDTM** [comment]
The toxins example aligns more with dissection than your method, since it's time investment/resources spent exclusively for research as the bomb is blown into the testing range instead of being used.

**Comment by ExcessiveUseOfCobblestone** [comment]
The people who do this toxins hooplah consider the point-gen aspect their job per the feedback i've received. They simply have no idea what else to do with the bombs.

**Comment by XDTM** [comment]
>They simply have no idea what else to do with the bombs.

Which is an underlying problem in toxins more than a design decision.

As far as i've seen in current techweb point generation methods, you need to actively work towards it in some way, and spend either time or resources to make it happen. Aside from toxins, we have the rad collectors, which require you to set them up and forgo a portion of generated power, and the ballmer peak, for what it's worth, still requires a balancing effort to maintain.

Adding points to normal surgery requires zero effort or cost in addition to the surgery itself, which goes against that design direction.
It's not a bad idea per se, if properly balanced, but i'd rather have departments have the option to go the extra mile for major research gains, rather than having minor research gains from doing business as usual without any decision-making involved.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm of the personal opinion that what you choose to use the points on should be the bottleneck, not "I forgo the job for research" AND what you choose to use the points on. It's hard to say RND is no longer a job when really we're giving every department a RND job that requires them to sit and do the thing for points.

I'm not sure of any other game that is meant to be played for 1-2 hours that requires you to devote your entire gameplay into boosting research. Closest I can think of is Civ 5 on quick/short and that's 1) a RTS where the units devote their time, you just tell them to do it and 2) passive gen that increases as time goes on (to varying degrees based on what you build/etc. although you're not halted from performing other actions altogether). Haven't played 6.

**Comment by ExcessiveUseOfCobblestone** [comment]
An alternative could be that I can balance these numbers to be a lesser variant of the experimental surgery, while still having the experimental surgery give boosts byond what normal surgery completion can provide.

**Comment by XDTM** [comment]
I agree that devoting full time to research is bad as well, since at that point you're not doing the department job in the first place. Dissections are an extra you can do if you have free time for bonus points, and since the doctor job tends to have high and low tides of work depending on the violence on station, they can use the low tides to do something instead of <s>erping</s> wandering in boredom.

Plus is works as a sort of bounty system now, where you get far more points for rarer specimen; that way it's more of an opportunity you can take rather than a full-time dedication to research-boosting.

Having surgery research plus dissection is definitely viable, although i'd rather have relatively low numbers. If research eventually branches into different types of points, surgery could be a great source of biology points, and making the gain more reliable would reinforce that.

**Comment by ExcessiveUseOfCobblestone** [comment]
Yep, ill do that too soon [tm]

**Comment by Tlaltecuhtli** [comment]
are these surgeries a 1 time thing or can they be repeated for the same amount of research?

**Comment by ExcessiveUseOfCobblestone** [comment]
repeatable because surgery sucks

File: code/modules/surgery/lipoplasty.dm [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	name = "lipoplasty"
 	steps = list(/datum/surgery_step/incise, /datum/surgery_step/clamp_bleeders, /datum/surgery_step/cut_fat, /datum/surgery_step/remove_fat, /datum/surgery_step/close)
 	possible_locs = list(BODY_ZONE_CHEST)
+	research_completion_bonus = 500 // biggest... winner?
```
Comment (Cyberboss) [inline code/modules/surgery/lipoplasty.dm]: ha

**Review by Cyberboss — APPROVED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Figure out if there's a surgery computer nearby. Have it say the amount of research points gained

**Comment by Barhandar** [comment]
>Aside from toxins, we have the rad collectors, which require you to set them up and forgo a portion of generated power

And produce jack shit research points while requiring a full tank of tritium (read: needs quite a bit of time + effort to get) and getting through it in five minutes or so.

**Comment by lolman360** [comment]
trit is easy as fuck to get as engineering but the passive pointgen from it is not worth wasting trit on when there's fusion/maxcap STBs

**Comment by ExcessiveUseOfCobblestone** [comment]
Closed until I have the desire to work on it again. Feel free to steal.

**Pull request closed without merging**